### PR TITLE
Use FileNameGroups for load dataset

### LINKED ISSUE #1219: When loading files like NAME_001.tif NAME_002.tif don't include NAME_180deg.tif
### Summary

The IAEA_Cylinder_WhiteBeam dataset has a sample folder with files named

`IMAT00006388_PSI_cylinder_Sample_000.tif` -> `IMAT00006388_PSI_cylinder_Sample_943.tif`

plus
```
IMAT00006388_PSI_cylinder_Sample_180deg.tif
IMAT00006388_PSI_cylinder_Sample_360deg.tif
```
Currently the 180deg and 360deg get included in the main stack of projections

We should probably audit all the variations of how files are being named.

### Steps To Reproduce

Data set is at `\\isis\shares\IMAT\ExampleData\IAEA_Cylinder_WhiteBeam\Sample\`

or Spheres_1219.tar.xz in the onedrive example data folder.

Load the Tomo folder with Load Dataset

In the main windows scroll though the image stack, the issue causes the `Spheres_Tomo_180deg.tiff` to be included as the 180th projection in the main sample stack. So as you scroll passed z=180 the rotation will jump.

### Expected Behaviour

180 deg should be loaded as a 180. 360 should be ignored

### Current Behaviour

The 'deg' files are included in the main stack, and create jumps

### Context

Current main, not yet tested on older versions

### PR DESCRIPTION
**Pull request by samtygier-stfc**
### Issue

Nearly there on #1219 

### Description

Move Load Dataset to new code. This covers the loading side, finding files in ImageLoadDialog still uses old methods.

Remove unused code

### Testing & Acceptance Criteria 
Check that you can load  with Load Dataset

### Documentation

Not yet

### CODE REVIEW

**Review by JackEAllen — CHANGES_REQUESTED** [review]
From an initial pass through, looks really good and changes work as expected! Thank you for the changes! Definitely on the larger side of what I'm happy to consider as one PR. If it was much larger I would suggest breaking into multiple PRs. I have left a few nitpicks here and there that I would like to get your opinion on. I'll give this another pass through to make sure I'm not missing anything before approving.

File: mantidimaging/core/io/test/filenames_test.py [hunk 1]
```diff
@@ -57,6 +57,13 @@ def test_pattern_match_different_digits(self):
         self.assertTrue(p1.match("img_1000.tif"))
         self.assertFalse(p1.match("img_0000.tif"))
 
+    def test_pattern_match_different_digits_no_zeros(self):
```
[the remark this answers is not in the document]
Comment (samtygier-stfc) [reply mantidimaging/core/io/test/filenames_test.py:60]: I think its simpler to do it in a single test here, as there is not lots of duplicated setup.

File: mantidimaging/gui/windows/image_load_dialog/test/presenter_test.py [hunk 2]
```diff
@@ -233,77 +233,68 @@ def test_ensure_sample_log_consistency_exits_when_none_or_empty_str(self, mock_l
...
+        sample_path = "/sample/tomo/tomo_0001.tiff"
+        sample_file = "tomo_0001.tiff"
+        sample_log_path = "/sample/tomo.log"
+
+        self.fields["Sample"].path_text.return_value = sample_path
+        self.fields["Sample"].file.return_value = sample_file
+        self.fields["Sample"].use.isChecked.return_value = True
+        self.fields["Sample Log"].path_text.return_value = sample_log_path
+
+        # sample_indices = [1, 1] #TODO
```
Comment (JackEAllen) [inline mantidimaging/gui/windows/image_load_dialog/test/presenter_test.py:247]: Did you mean to leave this in?

Comment (samtygier-stfc) [reply mantidimaging/gui/windows/image_load_dialog/test/presenter_test.py:247]: It gets fixed in the next PR #1728 ( ba2b7bb6d83889c53a268534796fc9aef215a19e )

**Review by JackEAllen — APPROVED** [review]
Thanks for making the small changes to add docstrings. Though they may be simple descriptions, they are enough to provide context for future usage should anyone else ever need to touch this. Happy to approve this 👍

**Pull request merged**